thrust/detail/allocator/allocator_traits.inl

```diff
@@ -246,7 +246,7 @@ __host__ __device__
     system(Alloc &)
 {
   // return a copy of a default-constructed system
-  typename allocator_system<Alloc>::type result;
+  typename allocator_system<Alloc>::type result{};
```

Comment:
We can't do brace initialization here; this assumes C++11. Please modify this change to not depend on C++11.

Reply:
Done.  Please give it another try.